src/routes/routes.ts

```diff
@@ -1,33 +1,116 @@
+export const SAFE_SUBSECTION_SLUG = 'safeSubsection'
+export const SAFE_SUBSECTION_ROUTE = `${SAFE_SECTION_ROUTE}/:${SAFE_SUBSECTION_SLUG}`
+
+// URL: gnosis-safe.io/app/:[SAFE_ADDRESS_SLUG]/:[SAFE_SECTION_SLUG]/:[SAFE_SUBSECTION_SLUG]
+export type SafeRouteSlugs = {
+  [SAFE_ADDRESS_SLUG]?: string
+  [SAFE_SECTION_SLUG]?: string
+  [SAFE_SUBSECTION_SLUG]?: string
+}
 
-export const BASE_SAFE_ROUTE = `${ROOT_ROUTE}/safes/:safeAddress`
+// Routes independant of safe/network
+export const WELCOME_ROUTE = '/welcome'
+export const OPEN_ROUTE = '/open'
+export const LOAD_ROUTE = `/load/:${SAFE_ADDRESS_SLUG}`
 
+// [SAFE_SECTION_SLUG], [SAFE_SUBSECTION_SLUG] populated safe routes
 export const SAFE_ROUTES = {
-  ASSETS_BALANCES: `${BASE_SAFE_ROUTE}/balances`,
-  ASSETS_COLLECTIBLES: `${BASE_SAFE_ROUTE}/balances/collectibles`,
-  TRANSACTIONS: `${BASE_SAFE_ROUTE}/transactions`,
-  ADDRESS_BOOK: `${BASE_SAFE_ROUTE}/address-book`,
-  APPS: `${BASE_SAFE_ROUTE}/apps`,
-  SETTINGS_BASE_ROUTE: `${BASE_SAFE_ROUTE}/settings`,
-  SETTINGS_DETAILS: `${BASE_SAFE_ROUTE}/settings/details`,
-  SETTINGS_OWNERS: `${BASE_SAFE_ROUTE}/settings/owners`,
-  SETTINGS_POLICIES: `${BASE_SAFE_ROUTE}/settings/policies`,
-  SETTINGS_SPENDING_LIMIT: `${BASE_SAFE_ROUTE}/settings/spending-limit`,
-  SETTINGS_ADVANCED: `${BASE_SAFE_ROUTE}/settings/advanced`,
+  ASSETS_BALANCES: `${ADDRESSED_ROUTE}/balances`, // [SAFE_SECTION_SLUG] === 'balances'
+  ASSETS_BALANCES_COLLECTIBLES: `${ADDRESSED_ROUTE}/balances/collectibles`, // [SAFE_SUBSECTION_SLUG] === 'collectibles'
+  TRANSACTIONS: `${ADDRESSED_ROUTE}/transactions`,
+  ADDRESS_BOOK: `${ADDRESSED_ROUTE}/address-book`,
+  APPS: `${ADDRESSED_ROUTE}/apps`,
+  SETTINGS: `${ADDRESSED_ROUTE}/settings`,
+  SETTINGS_DETAILS: `${ADDRESSED_ROUTE}/settings/details`,
+  SETTINGS_OWNERS: `${ADDRESSED_ROUTE}/settings/owners`,
+  SETTINGS_POLICIES: `${ADDRESSED_ROUTE}/settings/policies`,
+  SETTINGS_SPENDING_LIMIT: `${ADDRESSED_ROUTE}/settings/spending-limit`,
+  SETTINGS_ADVANCED: `${ADDRESSED_ROUTE}/settings/advanced`,
+}
+
+export type SafeRouteParams = { shortName: string; safeAddress: string }
+
+const isValidShortChainName = (shortName: string): boolean => {
+  if (!shortName) return false
+  const shortNames = getNetworks().map(({ shortName }) => shortName)
+  return shortNames.includes(shortName)
+}
+
+// Due to hoisting issues, these functions should remain here
+export const getPrefixedSafeAddressFromUrl = (): SafeRouteParams => {
+  const currentChainShortName = getCurrentShortChainName()
+
+  const match = matchPath<SafeRouteSlugs>(history.location.pathname, { path: ADDRESSED_ROUTE })
+
+  const prefixedSafeAddress = match?.params?.[SAFE_ADDRESS_SLUG]
+  const parts = prefixedSafeAddress?.split(':')?.filter(Boolean)
+
+  if (!prefixedSafeAddress || !parts?.length) {
+    return {
+      shortName: currentChainShortName,
+      safeAddress: '',
+    }
+  }
+
+  const shortName = parts[0]
```

Comment:
This won't support old-style bare addresses, will it?
Maybe `const shortName = parts.length === 2 ? parts[0] : currentChainShortName`?